### PR TITLE
Feat/upgrade simple proposal

### PR DESCRIPTION
**Pull request by BarbaraOliveira13**
🎩 Description
Update simple_proposal

**ADMIN PART**: Adding 2 new fields as checkboxes ('Scope' and 'Category') in the Backoffice when accessing the decidim-proposals edit/create form

**USER PART**: If the admin wants to make it mandatory, display the 'Scope' and 'Category' fields as required in the decidim-proposal creation form

📌 Related Issues
[Notion card](https://www.notion.so/opensourcepolitics/Lyon-Am-lioration-simple-proposal-7125574cf8374bfb8547eacefd78b549?pvs=4)

Testing
**ADMIN PART**
1. As admin go to dashboard
2. Go to a proposal
3. Go to components
4. For the #edit: Click to proposals settings (the gear icon ⚙️ named 'configure') OR, 
5. For the #create: click to 'add component/proposals' button (and check "Participants can create proposals")
6. At the end of 'Global settings' there are the two checkboxes for scope & category
Capture d’écran 2024-05-14 à 10 13 27

_____________

**USER PART**

1. Go to Processes
2. Click on a process 
3. Click to “more…” and choose “proposals”
4. Click to “new idea”
6. Scope & category fields are necessary, depending on the admin’s choice
Capture d’écran 2024-05-15 à 16 20 59

### CODE REVIEW

File: config/locales/fr.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+fr:
+  decidim:
+    components:
+      proposals:
+        settings:
+          global:
+            require_category: La catégorie est obligatoire
+            require_scope: Le secteur est obligatoire
```
Comment (Quentinchampenois) [inline config/locales/fr.yml:8]: I think you forgot to add french translations for `form` keys

File: db/migrate/20240519110747_add_require_fields_to_proposals.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddRequireFieldsToProposals < ActiveRecord::Migration[6.1]
```
Comment (Quentinchampenois) [inline db/migrate/20240519110747_add_require_fields_to_proposals.rb]: This migration is no longer needed

File: spec/system/create_proposal_spec.rb [hunk 3]
```diff
@@ -34,13 +36,10 @@ def visit_component
...
-  context "when category and scope are required" do
-    before do
-      allow(Decidim::SimpleProposal).to receive(:require_category).and_return(true)
-      allow(Decidim::SimpleProposal).to receive(:require_scope).and_return(true)
-    end
+  context "when category and scope are required," do
+    let(:settings) { { require_category: true, require_scope: true } }
 
-    context "without any scopes or categories" do
+    context "without any scopes or categories," do
```
Comment (Quentinchampenois) [inline spec/system/create_proposal_spec.rb]:
```suggestion
    context "without any scopes or categories" do
```

There is a typo here

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
I have few suggestions otherwise it looks good to me ! 🥇

**Review by luciegrau — APPROVED** [review]
Hi @BarbaraOliveira13 thank you for the work! I tried it locally and it seems to work great. 
@simonaszilinskas could you do a product review on your own to ensure that the PR responds to the initial request? Thank you!

**Review by Stef-Rousset — APPROVED** [review]
looks good to me, congrats !

File: config/locales/fr.yml (hunk 1 above)
Comment (BarbaraOliveira13) [reply config/locales/fr.yml:8]: Thanks, I verified and there is no need for the en.yml part. I used it unnecessarily when I tryed to backport the corresponding view. So I will delete the useless part!

Capture d’écran 2024-05-21 à 16 58 31

File: db/migrate/20240519110747_add_require_fields_to_proposals.rb (hunk 2 above)
Comment (BarbaraOliveira13) [reply db/migrate/20240519110747_add_require_fields_to_proposals.rb]: Are you sure ? When I deleted it (around last friday), I had an error "like" "raise UnknownAttributeError (...)require_scope(...)" when I tried to create a new proposal localy.

**Review by Quentinchampenois — APPROVED** [review]
LGTM ! 👍

**Pull request merged**